### PR TITLE
Fix notification shows artifact vector drawable issue #1952

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Fix issue #1952:  Android M shows roundabout maneuver icon artifact in the foreground notification.
The problem was we should use AppCompatDrawable for vectors on M or prior to M.
We updated the maneuver icon resources for the notification's RemoteView. When it's a vector drawable and as @Guardiola31337 mentioned in #1952 that some Arc curves icon will show artifact.

The way this PR fix it is: get the AppCompatDrawable and create a bitmap for the maneuver icon then apply to the notification's RemoteView to update.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Screenshots or Gifs

**Before fix on Nexus 5 OS 6**
before fix on Nexus 5 OS 6

**After fix on Nexus 5 OS 6**
after fix on Nexus 5 OS 6

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt [hunk 1]
```diff
@@ -302,17 +305,36 @@ class MapboxTripNotification constructor(
...
             currentManeuverId = maneuverImageId
-            collapsedNotificationRemoteViews?.setImageViewResource(
-                R.id.maneuverImage,
-                maneuverImageId
-            )
-            expandedNotificationRemoteViews?.setImageViewResource(
-                R.id.maneuverImage,
-                maneuverImageId
-            )
+            when (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
```
Comment (RingerJK) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]: Logic is duplicating, may be better to move it to one place, wdyt?

Comment (Guardiola31337) [reply libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]: I believe was duplicated on purpose to include the fix in both the legacy `libandroid-navigation` and `1.0` (`libtrip-notification`). So not a problem as `libandroid-navigation` code will be replaced when releasing `1.0`.

**Comment by JunDai** [comment]
@Guardiola31337 - does the CI check only auto-run against the origin repo?

**Comment by Guardiola31337** [comment]
@JunDai 
> does the CI check only auto-run against the origin repo?

That's correct. External contributions don't fire off CI, we (maintainers) need to pull the changes into a branch and push to the repo. So I recommend you to push directly to the repo creating a branch based off `master` and not a fork, it's going to be faster 😅

**Pull request closed without merging**